### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Andriykooo**
[DEMO LINK](https://Andriykooo.github.io/react_movies-list-add-form/index.html)

### CODE REVIEW

File: src/components/NewMovie/NewMovie.js [hunk 1]
```diff
@@ -1,13 +1,175 @@
...
+            placeholder="title"
+            name="title"
+            value={title}
+            onChange={this.onInputChange}
+            className={`new-movie__input 
+              ${fiedsError.titleError ? 'new-movie__input--error' : ''}`}
+          />
+          {fiedsError.titleError
+            && <span className="new-movie__error">Enter valid title!</span>}
+          <input
```
Comment (victor-arutiunov) [inline src/components/NewMovie/NewMovie.js:119]: I think that textarea will be better for description.

File: src/components/NewMovie/NewMovie.js [hunk 2]
```diff
@@ -1,13 +1,175 @@
...
+  onSubmit = (event) => {
+    const {
+      title,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    } = this.state;
+
+    // eslint-disable-next-line
+    const validUrlRegexp = /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@.\w_]*)#?(?:[.!/\\\w]*))?)$/g;
```
Comment (victor-arutiunov) [inline src/components/NewMovie/NewMovie.js:43]: You can use one regex variable for both of URL's.

File: src/components/NewMovie/NewMovie.js [hunk 3]
```diff
@@ -1,13 +1,175 @@
...
+            onChange={this.onInputChange}
+            className={`new-movie__input
+              ${fiedsError.imdbIdError ? 'new-movie__input--error' : ''}`}
+          />
+          {fiedsError.imdbIdError
+            && <span className="new-movie__error">Enter valid imdbId!</span>}
+          <button
+            type="submit"
+            className="new-movie__submit-btn"
+            disabled={Object.values(fiedsError).some(value => value === true)}
```
Comment (victor-arutiunov) [inline src/components/NewMovie/NewMovie.js:163]: Nice!

File: src/components/NewMovie/NewMovie.js [hunk 4]
```diff
@@ -1,13 +1,175 @@
...
+    imdbId: '',
+    fiedsError: {
+      titleError: false,
+      imgUrlError: false,
+      imdbUrlError: false,
+      imdbIdError: false,
+    },
+  };
+
+  onInputChange = (event) => {
```
Comment (victor-arutiunov) [inline src/components/NewMovie/NewMovie.js:20]: Cool!

File: src/components/NewMovie/NewMovie.js [hunk 5]
```diff
@@ -1,13 +1,175 @@
...
+    const { target } = event;
```
Comment (VinnieJ-2k20) [inline src/components/NewMovie/NewMovie.js:21]:
```suggestion
    const { name, value } = event.target;
```

File: src/components/NewMovie/NewMovie.js (hunk 2 above)
Comment (VinnieJ-2k20) [reply src/components/NewMovie/NewMovie.js:43]: Agree

File: src/components/NewMovie/NewMovie.js [hunk 6]
```diff
@@ -1,13 +1,175 @@
...
+
+    if (!validUrlRegexp.test(imgUrl)) {
+      return;
+    }
+
+    if (!validbUrlRegexp.test(imdbUrl)) {
+      return;
+    }
+
+    this.props.addMovie({ ...this.state });
```
Comment (VinnieJ-2k20) [inline src/components/NewMovie/NewMovie.js:75]: Don't pass whole state object here, craft separate one for the movie

File: src/components/NewMovie/NewMovie.js [hunk 7]
```diff
@@ -1,13 +1,175 @@
...
+          className="new-movie__form"
+        >
+          <input
+            type="text"
+            placeholder="title"
+            name="title"
+            value={title}
+            onChange={this.onInputChange}
+            className={`new-movie__input 
+              ${fiedsError.titleError ? 'new-movie__input--error' : ''}`}
```
Comment (VinnieJ-2k20) [inline src/components/NewMovie/NewMovie.js:115]: Use `classnames` library for this

File: src/components/NewMovie/NewMovie.js [hunk 8]
```diff
@@ -1,13 +1,175 @@
...
+      imgUrl: '',
+      imdbUrl: '',
+      imdbId: '',
+      fiedsError: {
+        titleError: false,
+        imgUrlError: false,
+        imdbUrlError: false,
+        imdbIdError: false,
+      },
+    });
```
Comment (VinnieJ-2k20) [inline src/components/NewMovie/NewMovie.js:89]: Let's move form reset to separate method.

File: src/components/NewMovie/NewMovie.js [hunk 9]
```diff
@@ -1,13 +1,175 @@
...
+      return;
+    }
+
+    if (!validUrlRegexp.test(imgUrl)) {
+      return;
+    }
+
+    if (!validbUrlRegexp.test(imdbUrl)) {
+      return;
+    }
```
Comment (VinnieJ-2k20) [inline src/components/NewMovie/NewMovie.js:73]: You use similar logic here and in lines 52-55. Move it to separate method (e.g. validate form).

**Review by VinnieJ-2k20 — APPROVED** [review]
I'll approve because basic task is completed correctly.

Regarding advanced task - it needs some improvement.
Besides things mentioned in the comments:

1. Control values need to be validated on blur (when focus is removed from the input)
2. Add button should be disabled if form has errors or unvalidated fields. 
3. Also, submit on pressing Enter should also be disabled in this case.

If you will push changes for the advanced task - ping mentors in your channel for review.

**Pull request closed without merging**